app/assets/javascripts/image-upload-markdown.js

```diff
@@ -0,0 +1,60 @@
+// This is a modified version of https://github.com/stuartnelson3/blog/blob/master/public/js/script.js
+
+
+$(document).on('drop', '.js-markdown-image-upload', function(e) {
+
+  e.preventDefault();
+  var dt = e.originalEvent.dataTransfer;
+  for (var i = 0; i < dt.files.length; i++) {
+    var file = dt.files[i];
+    var textarea = e.currentTarget;
+    handleFileUpload(textarea, file);
+  }
+
+  function addImageMarkdownPlaceholder(e, filename) {
+    var startPos = e.selectionStart;
+    var endPos = e.selectionEnd;
+    e.value = e.value.substring(0, startPos)
+      + "![Uploading " + filename + "...]()\r\n"
+      + e.value.substring(endPos, e.value.length);
+  }
+
+  function replaceImageMarkdownPlaceholder(textarea, value, filename){
+    var re = new RegExp("!\\[Uploading " + filename + "\\.\\.\\.\\]\\(\\)");
```

Comment:
This doesn't need regex.  Normal string replace will work and be a lot less ugly.